plugin.js

```diff
@@ -281,24 +286,64 @@ function processWorklets(t, path, processor) {
   }
 }
 
+const PLUGIN_BLACKLIST_NAMES = [
+  // '@babel/plugin-transform-destructuring', do not blacklist
+  'metro-react-native-babel-preset/node_modules/@babel/plugin-transform-object-assign',
+];
+
+const PLUGIN_BLACKLIST = PLUGIN_BLACKLIST_NAMES.map((pluginName) => {
+  try {
+    const blacklistedPluginObject = require(pluginName);
+    const blacklistedPlugin = blacklistedPluginObject.default({
+      assertVersion: (x) => true,
+    });
+
+    visitors.explode(blacklistedPlugin.visitor);
+    return blacklistedPlugin;
+  } catch (e) {
+    console.log(`No ${pluginName}!`);
+  }
+});
+
+function removePluginsFromBlacklist(plugins) {
+  PLUGIN_BLACKLIST.forEach((blacklistedPlugin) => {
+    if (!blacklistedPlugin) {
+      return;
+    }
+
+    let toRemove = [];
+    for (let i = 0; i < plugins.length; i++) {
+      if (
+        JSON.stringify(Object.keys(plugins[i].visitor)) !=
```

Comment:
shouldn't those objects just be the same? we load plugins using require and I think there should always be one source where plugins are loaded, no?

Reply:
Not true. Plugins can be loaded by several presets. Moreover, "require" returns a function, not an object.

Reply:
I wish we'd used soem different approach than JSON serialization. There is nothing technically wrong with it but it feels like the plugin object should have some field that we can extract its name from? If we want to use this as a last resort there should be a proper comment explaining that plugin objects are created by babel internals and they don't carry any identifier etc...